packages/contracts/contracts/minter-suite/Minters/MinterSetPricePolyptychERC20V5.sol

```diff
@@ -836,15 +844,27 @@ contract MinterSetPricePolyptychERC20V5 is
             projectId: projectId,
             coreContract: coreContract
         });
-        // @dev revert occurs during payment split if ERC20 token is not
-        // configured (i.e. address(0)), so check is not performed here
+
+        // get the currency address configured on the project
+        {
+            (address configuredCurrencyAddress, ) = SplitFundsLib
+                .getCurrencyInfoERC20(projectId, coreContract);
+            // validate that the currency address and symbols matches the project configured currency
+            require(
+                currencyAddress == configuredCurrencyAddress,
+                "Currency addresses must match"
+            );
+        }
 
         // validate that the specified maximum price is greater than or equal to the price per token
         require(
             maxPricePerToken >= pricePerTokenInWei,
             "Only max price gte token price"
         );
 
+        // @dev revert occurs during payment split if ERC20 token is not
+        // configured (i.e. address(0)), so check is not performed here
```

Comment:
nit: might makes sense to move this up to immediately after the currency address check comment

Reply:
updated in [41e4959](https://github.com/ArtBlocks/artblocks-contracts/pull/1141/commits/41e4959305fcb226e09b0aae86cc6c0b5860582f)